frontend/catalyst/jax_tracer.py

```diff
@@ -462,7 +463,12 @@ def _bind_native_controlled_op(qrp, op, controlled_wires, controlled_values):
         return qrp
 
     qrp = QRegPromise(qreg)
-    for op in device.expand_fn(quantum_tape):
+    if isinstance(device, qml.Device):
+        ops = device.expand_fn(quantum_tape)
```

Comment:
Why do we call `expand_fn` in the new version explicitly here? Can't we go with the same `QJITDevice.default_expand_fn` that we have?

Reply:
`qml.Device` is the old version of devices, it defines an `expand_fn`.

`qml.devices.Device` is the new device API and there is no expansion, what you need to do is to add transforms to the preprocess.

Reply:
Just clarification, we are moving to supporting both versions now with this commit, correct? Instead of supporting only `qml.devices.Device`?

Reply:
Yes we support both exactly, the previous version can be removed when lightning supports the new device API.